Algorithm.Framework/Risk/TrailingStopRiskManagementModel.cs

```diff
@@ -0,0 +1,89 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Algorithm.Framework.Portfolio;
+
+namespace QuantConnect.Algorithm.Framework.Risk
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IRiskManagementModel"/> that limits the relative drawdown
+    /// per holding to the specified percentage
+    /// </summary>
+    public class TrailingStopRiskManagementModel : RiskManagementModel
+    {
+        private readonly decimal _maximumDrawdownPercent;
+        private Dictionary<Symbol, decimal> _trailingHighs = new Dictionary<Symbol, decimal>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TrailingStopRiskManagementModel"/> class
+        /// </summary>
+        /// <param name="maximumDrawdownPercent">The maximum percentage relative drawdown allowed for any single security holding, defaults to 5% drawdown per security</param>
+        public TrailingStopRiskManagementModel(decimal maximumDrawdownPercent = 0.05m)
+        {
+            _maximumDrawdownPercent = -Math.Abs(maximumDrawdownPercent);
+        }
+
+        /// <summary>
+        /// Manages the algorithm's risk at each time step
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="targets">The current portfolio targets to be assessed for risk</param>
+        public override IEnumerable<IPortfolioTarget> ManageRisk(QCAlgorithmFramework algorithm, IPortfolioTarget[] targets)
+        {
+            foreach (var kvp in algorithm.Securities)
+            {
+                var symbol = kvp.Key;
+                var security = kvp.Value;
+
+                // Remove if not invested
+                if (!security.Invested)
+                {
+                    if (_trailingHighs.ContainsKey(symbol))
+                    {
+                        _trailingHighs.Remove(symbol);
+                    }
+                    continue;
+                }
+
+                // Add newly invested securities
+                if (!_trailingHighs.ContainsKey(symbol))
+                {
+                    _trailingHighs.Add(symbol, security.Holdings.AveragePrice); // Set to average holding cost
+                    continue;
+                }
+
+                // Check for new highs and update - set to tradebar high
+                if (_trailingHighs[symbol] < security.High)
+                {
+                    _trailingHighs[symbol] = security.High;
+                    continue;
+                }
+
+                // Check for securities past the drawdown limit
+                var securityHigh = _trailingHighs[symbol];
+                var drawdown = (security.Low / securityHigh) - 1m;
```

Comment:
Is comparing against `security.Low` correct here? Consider we're using daily data and there's a really bad intraday low, but we close the day higher? I'm thinking it makes more sense to compare against the current `security.Price`  -- thoughts?

Reply:
Yes, I thought about this and because it's a risk management model I though it best to cover worst case scenario. I agree that the past data is also not really relevant, so in that case `security.Price` makes sense. Should I make this change?

Reply:
Please keep it as low; its _triggering_ the trailing stop; so is still relevant. The delay in the application is a fault of using long-term data like that. We shouldn't accept higher drawdowns because of choice of bars?

An extreme case would be weekly bars; during the flash crash week. I would expect a trailing stop to trigger in this case?